### PR TITLE
[SMALLFIX] More test coverage in TachyonMasterFaultToleranceIntegrationTest

### PR DESCRIPTION
**Pull request by uronce-cc**

### CODE REVIEW

File: clients/unshaded/src/main/java/tachyon/hadoop/AbstractTFS.java [hunk 1]
```diff
@@ -382,7 +382,6 @@ public FileStatus getFileStatus(Path path) throws IOException {
    * @see org.apache.hadoop.fs.FileSystem#createFileSystem(java.net.URI,
    *      org.apache.hadoop.conf.Configuration)
    */
-  @Override
```
Comment (gpang) [inline clients/unshaded/src/main/java/tachyon/hadoop/AbstractTFS.java]: why is this removed?

File: integration-tests/src/test/java/tachyon/master/MasterFaultToleranceIntegrationTest.java [hunk 2]
```diff
@@ -115,6 +116,35 @@ public void faultTest() throws IOException {
...
 
   @Test
+  public void deleteFileFaultTest() throws Exception {
+    // Kill leader -> create files -> kill leader -> delete files, repeat.
+    List<Pair<Long, TachyonURI>> answer = Lists.newArrayList();
+    for (int kills = 0; kills < MASTERS - 1; kills ++) {
+      Assert.assertTrue(mLocalTachyonClusterMultiMaster.killLeader());
+      CommonUtils.sleepMs(Constants.SECOND_MS * 2);
+
+      if (kills % 2 != 0) {
```
Comment (gpang) [inline integration-tests/src/test/java/tachyon/master/MasterFaultToleranceIntegrationTest.java]: Both parts of the branch should have a comment on what it will be doing. (one is deleting files, the other is creating files?)

**Comment by gpang** [comment]
I have some minor comments.

**Comment by calvinjia** [comment]
@uronce-cc could you resolve the merge conflicts please? Also, do these versions work on the hdfs profile? It seems like the master fault tolerance suite as a whole is not working well there.

File: clients/unshaded/src/main/java/tachyon/hadoop/AbstractTFS.java (hunk 1 above)
Comment (uronce-cc) [inline clients/unshaded/src/main/java/tachyon/hadoop/AbstractTFS.java]: It blocks compilation.

**Comment by uronce-cc** [comment]
Conflict has been resolved, but all cases fail in hdfs1Test profile, I'll try to address that in a follow up PR.

**Comment by apc999** [comment]
Jenkins, test this please.

File: integration-tests/src/test/java/tachyon/master/MasterFaultToleranceIntegrationTest.java [hunk 3]
```diff
@@ -122,7 +123,42 @@ public void faultTest() throws IOException, TException {
...
+        faultTestDataCheck(answer);
+
+        faultTestDataCreation(new TachyonURI(PathUtils.concatPath(
+            TachyonURI.SEPARATOR, "data_" + kills)), answer);
+        faultTestDataCheck(answer);
+      }
+    }
+  }
+
+  // TODO: Resolve the issue with HDFS as UnderFS
```
Comment (jsimsa) [inline integration-tests/src/test/java/tachyon/master/MasterFaultToleranceIntegrationTest.java]: style: // TODO(cc): Resolve the issue with HDFS as UnderFS.

(tag your TODOs and end sentences with a period)

File: integration-tests/src/test/java/tachyon/master/MasterFaultToleranceIntegrationTest.java [hunk 4]
```diff
@@ -137,4 +173,32 @@ public void createFilesTest() throws IOException, TException {
       Assert.assertEquals(TachyonURI.SEPARATOR + k, files.get(k));
     }
   }
+
+  // TODO: Resolve the issue with HDFS as UnderFS
```
Comment (jsimsa) [inline integration-tests/src/test/java/tachyon/master/MasterFaultToleranceIntegrationTest.java]: ditto

**Comment by jsimsa** [comment]
Thank you for the tests. Just a couple of minor comments.

File: integration-tests/src/test/java/tachyon/master/MasterFaultToleranceIntegrationTest.java (hunk 3 above)
Comment (uronce-cc) [inline integration-tests/src/test/java/tachyon/master/MasterFaultToleranceIntegrationTest.java]: done

File: integration-tests/src/test/java/tachyon/master/MasterFaultToleranceIntegrationTest.java (hunk 4 above)
Comment (uronce-cc) [inline integration-tests/src/test/java/tachyon/master/MasterFaultToleranceIntegrationTest.java]: done

File: integration-tests/src/test/java/tachyon/master/MasterFaultToleranceIntegrationTest.java (hunk 2 above)
Comment (uronce-cc) [inline integration-tests/src/test/java/tachyon/master/MasterFaultToleranceIntegrationTest.java]: done

**Comment by gpang** [comment]
LGTM

**Comment by gpang** [comment]
Jenkins, test this please

**Comment by gpang** [comment]
Could you fix the merge conflicts? Thanks!

**Comment by jsimsa** [comment]
LGTM

**Comment by uronce-cc** [comment]
conflicts resolved.
